src/filtering/filterParser.ts

```diff
@@ -357,10 +360,20 @@ class BooleanTerm {
     } else if (this.factors.length === 1) {
       s += this.factors[0];
     } else {
-      s += '{"$and":[';
+      if (this.not) {
+        // since we're NOT-ing the underlying assertions in Predicate
+        // switch the AND to an OR using De Morgan's laws
+        s += '{"$or":[';
+      } else {
+        s += '{"$and":[';
+      }
       this.factors.forEach((f) => (s += f + ','));
       s = s.slice(0, -1);
-      s += ']}';
+      if (this.not) {
+        s += ']}';
+      } else {
+        s += ']}';
+      }
```

Comment:
Okay the same thing is happening here and since I'm struggling a little to figure out everything going on, are you sure this is correct? We don't want to accidentally have the opposite expression of what we're aiming for

Reply:
Ah, this was a hold over from when I was doing something different - I'll delete this